PythonAPI/examples/autoware_demo.py

```
@@ -163,7 +164,13 @@ def generate_imu_blueprint(blueprint_library):
 def generate_gnss_blueprint(blueprint_library):
     """Generates a blueprint for GNSS"""
 
-    blueprint = blueprint_library.find("sensor.other.gnss")
+    try:
+        blueprint = blueprint_library.find("sensor.other.autoware_gnss")
+        print('Spawned Autoware GNSS successfully')
+    except Exception:
+        traceback.print_exc()
+        log_warning('Autoware GNSS does not exist, reverting to native Carla GNSS')
+        blueprint = blueprint_library.find("sensor.other.gnss")
```

I can see why you did this in addition to hard-coding MGRS offset in the standard GNSS publisher, but I think in general this should just fail and not fallback to some hardcoded values useful only in shinjuku.

So in addition to my comment about hard-coded MGRS values in non-MGRS GNSS publisher member function, I think better behavior would be to not add MGRS in the non-MGRS publish and fail here.

---

Done